docs/modules/ROOT/pages/high_level/security_group_host_design.adoc

```diff
@@ -0,0 +1,583 @@
+With Delta update, message size with one security group rule will be around 150 bytes plus outer goal state message and protobuf overhead (around 50 bytes, message total ~200 bytes).
+Without Delta update, full message size with 100 security group rules will be 150 bytes times 100 plus outer goal state message and protobuf overhead (around 50 bytes, message total ~15,050 bytes).
+
+=== Remote Security Group Rule Handling
+
+A Security Group rule can refer to a remote security group for both ingress and egress traffic.
+When a security group rule is added to security group A, customer can set security group B as the source (for ingress rule) or destination (for egress rule).
+This allows ports associated with security group B to access ports that are port of security group A, and vice versa.
+
+[source,shell]
+------------------------------------------------------------
++-----------+     +-----------+ 
+| Compute   |     | Compute   |
+| Node 1    |     | Node 2    |
+|           |     |           |
+|       OVS + <---| OVS       |
+|           |     |           |
+| Port 1    |     | Port 2    |
+| SG A      |     | SG B      |
++-----------+     +-----------+
+------------------------------------------------------------
+
+The default security group has rules that allow associated ports of the default rule to talk to each other.
+
+To support this using the minimal set of OpenFlow rules, we will mark remote ports with its associated security group using conjunctive flows discussed in the next session.
+We will update our neighbor configuration schema to include associated security group IDs.
+For L3 neighbor, we will leverage on-demand rules so that the OpenFlow rule for the marking is only setup when needed (for egress).
+For L2 neighbor, we will go ahead to setup its OpenFlow rules for the marking since it is limited by the subnet size.
+We will consider doing on-demand rule for L2 neighbors in the future if needed.
+
+=== Conjunctive flows
+
+With a security group rule that is associated with a remote group ID, ingress rule will need to match on the "n" remote ports IP address (nw_src) and match on the mac addresses (dl_src) for all the ports "m" associated with this security group.
+It will result with n * m flows without using conjuctive flows.
+The situation is much worse when there are multiple (e.g. 10) security groups associated with a single port.
+
+We will use a conj_id for each combination of (remote SG_id, local port SG_id, ingress/egress, ethertype, flow priority).
+This rule is shared between security group rules and used by any flow that matches the same combo.
+
+Conjunctive flows can have 2 or more dimensions.
+We will use the first dimension to match the remote ports IP.
+We want to merge the rules with when there are multiple IP address if possible.
+The second dimension will be used to match the local port mac addresses and also other portion of the rule other than its remote SG_id.
```

Comment:
Do we match the MAC addresses of the local port or remote ports?

Reply:
we use the mac address to match the local port.